CorePayments/src/main/java/com/paypal/android/corepayments/API.kt

```diff
@@ -73,12 +75,20 @@ class API internal constructor(
         return clientID
     }
 
-    suspend fun sendAnalyticsEvent(name: String) {
-        try {
-            val clientID = fetchCachedOrRemoteClientID()
-            analyticsService.sendAnalyticsEvent(name, clientID)
-        } catch (e: PayPalSDKError) {
-            Log.d("[PayPal SDK]", "Failed to send analytics due to missing clientID: ${e.message}")
+    /**
+     * Sends analytics event to https://api.paypal.com/v1/tracking/events/ via a background task.
+     */
+    fun sendAnalyticsEvent(name: String) {
+        GlobalScope.launch {
```

Comment:
Do we need `CoroutineScope(dispatcher).launch(exceptionHandler)` here instead? I figured the `GlobalScope.launch` was sufficient since we don't really care about handling the result of the failure of analytics background tasks.

Reply:
[Android coroutines best practices](https://developer.android.com/kotlin/coroutines/coroutines-best-practices) says that we should avoid `GlobalScope`, because we are "hardcoding" the `Dispatcher`, and it might be dificult to test. 
Lets see what happens when we write the unit test, we might just end up doing another `CoroutineScope` with a dispatcher injected to it

Reply:
we could end up with something like this:
```kotlin
class API internal constructor(
    private val externalScope: CoroutineScope = GlobalScope,
    private val defaultDispatcher: CoroutineDispatcher = Dispatchers.Default
) {
    
    fun sendAnalticsEvent() {
        externalScope.launch(defaultDispatcher) {
            //send event
        }
    }
}
```

Reply:
Ok - we can 🏈 punt on in it until the unit test ticket (DTNOR-826)